src/main/frontend/state.cljs

```diff
@@ -951,16 +956,37 @@ Similar to re-frame subscriptions"
   (when-not (get-selection-start-block)
     (swap! state assoc :selection/start-block start-block)))
 
+(defn get-selection-blocks
+  []
+  (->> (:selection/blocks @state)
+       (remove nil?)))
+
+(defn should-select-block?
+  "Checks if the block belongs to the same page and container"
+  ([block]
+   (should-select-block? (first (get-selection-blocks)) block))
+  ([first-selected-block block]
+   (if first-selected-block
+     (let [container (util/get-block-container first-selected-block)
+           first-selected-block-id (get-block-id first-selected-block)
+           first-selected-block ((resolve 'frontend.db.model/get-block-by-uuid) first-selected-block-id)
```

Comment:
I appreciate trying to keep the diff smaller but circular deps like this makes the code harder to maintain, especially in a ns that is already messy. Moving this and dependent fns to `frontnend.util.block` or `frontend.handler.common.block` could work well. Happy to help do this cleanup on this or a followup PR

Reply:
I actually gave this a shot before using resolve. The problem is that multiple namespaces are heavily coupled with  `frontend.state` (in this case `frontend.db.model`). Moving some functions to `frontnend.util.block` makes sense though, so I did that anyway. I might be missing a trivial solution, so feel free to make a suggestion or push your own changes.